In [3]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [32]:
dataset = pd.read_csv("train.csv")
dataset.head()

##data = pd.read_csv('creditcard_2023.csv')

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

In [33]:
dataset.drop(['PassengerId','Name','Ticket'], axis=1, inplace=True)
dataset.head()

Survived  Pclass     Sex   Age  SibSp  Parch     Fare Cabin Embarked
0         0       3    male  22.0      1      0   7.2500   NaN        S
1         1       1  female  38.0      1      0  71.2833   C85        C
2         1       3  female  26.0      0      0   7.9250   NaN        S
3         1       1  female  35.0      1      0  53.1000  C123        S
4         0       3    male  35.0      0      0   8.0500   NaN        S

In [34]:
dataset.columns

Index(['Survived', 'Pclass', 'Sex', 'Age', 'SibSp', 'Parch', 'Fare', 'Cabin',
       'Embarked'],
      dtype='object')

### 2. Procentaje de nulos

In [35]:
dataset.isnull().mean()

Survived    0.000000
Pclass      0.000000
Sex         0.000000
Age         0.198653
SibSp       0.000000
Parch       0.000000
Fare        0.000000
Cabin       0.771044
Embarked    0.002245
dtype: float64

#### Eliminamos la columna Cabin dato que tiene 77.1% de faltantes 

In [36]:
dataset.columns

Index(['Survived', 'Pclass', 'Sex', 'Age', 'SibSp', 'Parch', 'Fare', 'Cabin',
       'Embarked'],
      dtype='object')

In [37]:
dataset.drop(['Cabin'], axis=1, inplace=True)

In [38]:
dataset.columns

Index(['Survived', 'Pclass', 'Sex', 'Age', 'SibSp', 'Parch', 'Fare',
       'Embarked'],
      dtype='object')

### Imputamos la media en la columna Age

In [39]:
media_age = dataset['Age'].mean()
media_age

29.69911764705882

In [40]:
media_age = dataset['Age'].mean()
dataset['Age'] = dataset['Age'].fillna(media_age).astype(int)

#### Imputamos embarked

In [41]:
dataset['Embarked'].value_counts()

Embarked
S    644
C    168
Q     77
Name: count, dtype: int64

In [42]:
dataset['Embarked'] = dataset['Embarked'].fillna('S')

#### 3. Codificación de variables categóricas

In [43]:
dataset.head()

Survived  Pclass     Sex  Age  SibSp  Parch     Fare Embarked
0         0       3    male   22      1      0   7.2500        S
1         1       1  female   38      1      0  71.2833        C
2         1       3  female   26      0      0   7.9250        S
3         1       1  female   35      1      0  53.1000        S
4         0       3    male   35      0      0   8.0500        S

In [44]:
dataset['Pclass'].value_counts()

Pclass
3    491
1    216
2    184
Name: count, dtype: int64

In [45]:
dataset['SibSp'].value_counts()

SibSp
0    608
1    209
2     28
4     18
3     16
8      7
5      5
Name: count, dtype: int64

In [46]:
dataset['Sex'].value_counts()

Sex
male      577
female    314
Name: count, dtype: int64

#### Codificacmos variable sex con One Hot Encodign

In [47]:
dataset['Sex'] = pd.get_dummies(dataset['Sex'], drop_first=True)

#### Codificamos variables Embarked con frecuency Encodign 

In [50]:
codificador_embarked = dataset['Embarked'].value_counts()

In [51]:
dataset['Embarked'] = dataset['Embarked'].map(codificador_embarked)

In [52]:
dataset.head()

Survived  Pclass    Sex  Age  SibSp  Parch     Fare  Embarked
0         0       3   True   22      1      0   7.2500       646
1         1       1  False   38      1      0  71.2833       168
2         1       3  False   26      0      0   7.9250       646
3         1       1  False   35      1      0  53.1000       646
4         0       3   True   35      0      0   8.0500       646

## 4. Estandarización de variables 

In [54]:
from sklearn.preprocessing import MinMaxScaler

In [61]:
mm_scaler = MinMaxScaler()

X = dataset.drop('Survived', axis=1)
scaled_feature = mm_scaler.fit_transform(X)
scaled_df = pd.DataFrame(scaled_feature, columns=X.columns)

In [62]:
scaled_df['Survived'] = dataset['Survived']
scaled_df.head()

Pclass  Sex     Age  SibSp  Parch      Fare  Embarked  Survived
0     1.0  1.0  0.2750  0.125    0.0  0.014151   1.00000         0
1     0.0  0.0  0.4750  0.125    0.0  0.139136   0.15993         1
2     1.0  0.0  0.3250  0.000    0.0  0.015469   1.00000         1
3     0.0  0.0  0.4375  0.125    0.0  0.103644   1.00000         1
4     1.0  1.0  0.4375  0.000    0.0  0.015713   1.00000         0

In [63]:
X

Pclass    Sex  Age  SibSp  Parch     Fare  Embarked
0         3   True   22      1      0   7.2500       646
1         1  False   38      1      0  71.2833       168
2         3  False   26      0      0   7.9250       646
3         1  False   35      1      0  53.1000       646
4         3   True   35      0      0   8.0500       646
..      ...    ...  ...    ...    ...      ...       ...
886       2   True   27      0      0  13.0000       646
887       1  False   19      0      0  30.0000       646
888       3  False   29      1      2  23.4500       646
889       1   True   26      0      0  30.0000       168
890       3   True   32      0      0   7.7500        77

[891 rows x 7 columns]

In [64]:
## scaled_df.to_csv('../data/processed/features_for_model.csv')
scaled_df.to_csv('features_for_model.csv')